src/app/utils/helpers.ts

```diff
@@ -43,3 +44,16 @@ export const isValidTxOasisHash = (hash: string): boolean => /^[0-9a-fA-F]{64}$/
 export const isValidTxEthHash = (hash: string): boolean => /^0x[0-9a-fA-F]{64}$/.test(hash)
 
 export const isValidTxHash = (hash: string) => isValidTxOasisHash(hash) || isValidTxEthHash(hash)
+
+export const getEthAccountAddress = (preimage: AddressPreimage | undefined): string | undefined => {
+  if (preimage?.context !== 'oasis-runtime-sdk/address: secp256k1eth') {
+    // We can only determine the ETH address if there was a preimage,
+    // and the generation context was secp256k1eth
+    return undefined
```

Comment:
> We have a large problem with other contexts scream

Well, everything I know comes from here: https://github.com/oasisprotocol/oasis-indexer/blob/main/api/spec/v1.yaml#L1577

How can I find more info about how to derive the ETH address in other contexts? (If possible at all.)

Reply:
this is failing when deriving oasis1 address from 0xa8E1A15b898df5B6807F034f7160cd6B8018937C

Reply:
> this is failing when deriving oasis1 address from 0xa8E1A15b898df5B6807F034f7160cd6B8018937C

As far as I can tell, the Oasis address we can derive from this ETH address is `oasis1qp0q8urx8qp7wa5qpg4pxpjc7f5xejklsy74ddta`.

Now if I check out this account [via the API](https://index-staging.oasislabs.com/v1/emerald/accounts/oasis1qp0q8urx8qp7wa5qpg4pxpjc7f5xejklsy74ddta), then I get this:
``` JSON
{
  "address":"oasis1qp0q8urx8qp7wa5qpg4pxpjc7f5xejklsy74ddta",
  "balances":[],
  "evm_balances":[],
  "stats":{
    "num_txns":0,
    "total_received":"0",
    "total_sent":"0"
  }
}
```

Note that there is no `address_preimage` field. Normal accounts (like [this one](https://index-staging.oasislabs.com/v1/emerald/accounts/oasis1qz670t637yyxshnlxhjj5074wgwl94d0x5x69zqd)) have ot.

So as far as I can tell, the reason we can't get back the ETH address is because of the API's behavior, isn't it?

Reply:
Discussed, and it does look like an API bug. sender_0_eth shouldn't be there. https://github.com/oasisprotocol/oasis-indexer/issues/365